functions/go/set-namespace/README.md

```diff
@@ -17,68 +17,114 @@ Namespaces are often used in the following scenarios:
 
 ## Usage
 
-This function can be used with any KRM function orchestrators (e.g. kpt).
-
-- If the resource is `Namespace`, `set-namespace` updates the `metadata.name` field.
-- If the resource is `RoleBinding` or `ClusterRoleBinding` resource, the function updates 
-  the namespace field in the `subjects` element whose name is `default`.
-- If the resource is `CustomResourceDefinition` (CRD), `set-namespace` updates the 
-  `spec/conversion/webhook/clientConfig/service/namespace` field.
-- If the resource is `APIService`, `set-namespace` updates the
-  `spec/service/namespace` field.
-- If there is a [`depends-on`] annotation for a namespaced resource, the namespace
-  section of the annotation will be updated if the referenced resource is also
-  declared in the package.
+This function replaces the KRM resources existing namespace to a new value.
 
+### Target KRM resources
+
+- This function updates all namespace-scoped KRM resources `metadata.namespace` fields. 
+  We determine whether a KRM resource is namespace scoped by checking if it has `metadata.namespace` set and matches the "oldNamespace"
+  If not, this function won't add new namespace.
+- This function updates `RoleBinding` and `ClusterRoleBinding` resources `subjects` element whose kind is `ServiceAccount`
+  and the subject's `namespace` is set and matches the "oldNamespace".
+- This function updates `CustomResourceDefinition` (CRD) `spec/conversion/webhook/clientConfig/service/namespace` field 
+  if the field is set and matches the "oldNamespace"
+- This function updates `APIService` `spec/service/namespace` field if the field is set and matches the "oldNamespace"
+- This function updates the KRM resources annotation `config.kubernetes.io/depends-on` if this annotation contains the 
+  matching namespace.
+
+### FunctionConfig
+
+This function supports the default `ConfigMap` as function config and a custom `SetNamespace`. See below examples
+
+ConfigMap as functionConfig
```

Comment:
Styling nit: `ConfigMap` should have backtick when it's a Kind name.

Similarly for `SetNamespace` and `Namespace`. Please audit this doc.

Reply:
done.